CMakeLists.txt

```diff
@@ -311,11 +311,12 @@ endif()
 
 # Add all local subdirecetories
 add_subdirectory(libzeth)
+add_subdirectory(libtool)
 
 # If zeth is being used as a dependency, skip the tools build
 if ("${IS_ZETH_PARENT}")
   add_subdirectory(prover_server)
-  add_subdirectory(verifier)
+  add_subdirectory(zeth_tool)
```

Comment:
What about naming it `zethctl`?

Reply:
I'm not sure what the `ctl` part means.  I think it should be something pretty intuitive.

Reply:
> I'm not sure what the ctl part means

Ah yes, ctl = control
(else `zethcli` or `zeth-cli`?)

`zeth-tool` is fine, simply not a huge fan of the "tool" suffix here - i.e. most of the time we talk about a command line interface, not a "tool" which sounds rather very generic.

Reply:
Ah I just saw your comment after pushing mine above (apparently needed to update the UI), yeah, that's what I thought (the collision with the client's cli)

Reply:
I had another look and `zeth_helper` should now be `zeth-helper` in the latest commit.

Reply:
> On balance, I'd probably prefer to avoid the cli in the name. It always seems superfluous (you know it's the cli if you are invoking it from the shell) 

Yes that's a good point.

> and even more generic than tool or util.

I think this is arguable - to me `cli` and `tool` are as generic tbh, but yes that's right `cli` isn't really a good prefix too I agree.

> The logic with adding a suffix here (instead of in the python client) was this the python client zeth is the main entry point for most administrator and client wallet operations, whereas this new tool is for more detailed operations.

Yes, I was surely not suggesting to add `tool` as prefix to the zeth wallet cli, I perfectly see why you would add it here.
The general point is that we start to have quite a few CLIs: the cpp one (the "tool"), the wallet one ("zeth") which in fact has more functionalities than just the "wallet", the helper one etc. I think a little bit of renaming wouldn't hurt here.
From what I can see, the "tool" (i.e. `zeth-tool`) is used to manipulate the proof objects with the subcommands:
```
        {"verify", verify_cmd},
        {"prove", prove_cmd},
        {"dump-proof", dump_proof_cmd},
```
So maybe we can rename the executable `zeth-proof` and the commands as `zeth-proof generate`, `zeth-proof verify`, `zeth-proof dump`. Of course, this won't work anymore if the tool is extended to carry out non-proof related work.

If we are short of names for now we can stick to "zeth-tool" no big deal at this stage really, but I'd suggest we add a README in https://github.com/clearmatics/zeth/tree/zeth-tool/zeth_tool to better explain the role of the executable.

> Regarding the zeth_helper, I've tried to use - consistently for command names, but IIRC python would not allow - for some reason. Although there may be a different way to add that command - I'll look into it.
>> I had another look and zeth_helper should now be zeth-helper in the latest commit.

Thanks. Here again no huge deal, but in general I prefer to avoid separators `_` and `-` in the names of the commands. Some commands use `_` some use `-` and I find it confusing because there aren't conventions about that (so you end up trying with an `-` and then figure out that it should be `_` after getting an error the first time. Clearer to avoid it all together when the name isn't too ugly)

Reply:
I agree `tool` is somewhat generic.  However, it does at least imply something a bit beyond the main functionality.  Not necessariily central to the primary operation, but for advanced or more special-case operations.

While the current functionality is mainly around proofs, etc, but including `proof` in th name may be a bit too specific.  The intention is that any other operations could be added here.  For example (as discussed in the past) it could make sense to have this tool perform the one-shot dummy keypair generation, rather than relying on the server doing it automatically if an existing keypair file does not exist.  As another example, if this tool had existed earlier, I would have added functionality to extract the PK and VK from a keypair file, to generate test data.  I'm expecting it to build up this kind of functionality, and possibly eventually be invoked by the main client tools where that makes sense.

I've added a brief readme to try and capture the intent.  However, I still can't think of anything better than `zeth-tool` for the name.

Reply:
> I've added a brief readme to try and capture the intent.

Great, thanks @dtebbs 

> However, I still can't think of anything better than zeth-tool for the name.

No big deal, the README documents the CLI's purpose. We can rename the "tool" if we find a better name in the future.